src/interface.jl

```diff
@@ -100,13 +121,57 @@ struct LogFuncDensity{F}
 end
 LogFuncDensity
 
-@inline isdensitytype(::Type{<:LogFuncDensity}) = true
+@inline hasdensity(::LogFuncDensity) = true
 
-@inline logdensityof(density::LogFuncDensity, x) = density._log_f(x)
-@inline logdensityof(density::LogFuncDensity) = density._log_f
+@inline logdensityof(d::LogFuncDensity, x) = d._log_f(x)
+@inline logdensityof(d::LogFuncDensity) = d._log_f
```

Comment:
I'm not sure if these `@inline` annotations should be added. Usually the heuristic of the compiler is good and I saw examples where it actually degraded performance.

Reply:
Well, the Julia compiler is pretty good at auto-inlining nowadays, but sometimes I've seen it fail in surprisingly clear situations, or seen an `@inline` help with return type inference. I'd like to leave it in - it can't hurt, I'd say, and I can't think of a situation in which I wouldn't want this to be inlined. Especially for `hasdensity` it might help the compiler elide code in `if hasdensity(...) ... else ... end` situations (it should of course do it by itself, but a little hinting can't be bad).